packages/facility-server/app/sync/pullIncomingChanges.js

```
@@ -63,3 +68,60 @@ export const pullIncomingChanges = async (centralServer, sequelize, sessionId, s
+  let records = [];
+  let writes = [];
+
+  const endpoint = () => `sync/${sessionId}/pull${fromId ? `?fromId=${fromId}` : ''}`;
+  stream: for await (const { kind, data } of centralServer.stream(endpoint)) {
+    if (records.length >= WRITE_BATCH_SIZE) {
+      // do writes in the background while we're continuing to stream data
+      writes.push(writeBatch(records));
+      records = [];
+    }
+
+    handler: switch (kind) {
```

Biome (1.9.4)

[error] Unexpected label.

Only loops should be labeled.
The use of labels for other statements is suspicious and unfamiliar.

(lint/suspicious/noConfusingLabels)